## Question 1

### (A).

$\frac{1500+1000}{1500+2250+1000+250} - \frac{875+1125}{375+875+2625+1125} = \frac{2500}{5000} - \frac{2000}{5000} =\frac{1}{10}$

### (B).

$\hat{ATE_{M}} = \frac{1500}{1500+2250}-\frac{875}{275+875} = \frac{1500}{3750}-\frac{875}{1150} = -\frac{83}{230} $

$\hat{ATE_{F}} = \frac{1000}{1000+250}-\frac{1125}{2625+1125} = \frac{1000}{1250}-\frac{1125}{3750} = \frac{1}{2} $

### (C).

There is much difference between male groups and female groups, and also both groups are different from the aggregate effect.

This is because we omitted the variable 'gender', so we see in the control group and treatment group, the gender distribution is not identical(Biased).

### (D).

The assumption is that gender distribution is completely irrelevant to assignment to treatment/control. In other words, the female and male proportions should be the same for both treatment and control

## Question 2

### (A).

In [1]:
school = read.csv("D:/Study/Business Analytics/LA/HW7/schooling.csv")
str(school)

'data.frame':	10000 obs. of  3 variables:
 $ LogEarning: num  12 9.16 10.41 13.04 10.06 ...
 $ College   : int  0 0 0 0 1 0 0 0 0 1 ...
 $ IQ        : int  82 79 93 87 103 77 88 96 89 109 ...


In [2]:
school$NormIQ = scale(school$IQ)
str(school)

'data.frame':	10000 obs. of  4 variables:
 $ LogEarning: num  12 9.16 10.41 13.04 10.06 ...
 $ College   : int  0 0 0 0 1 0 0 0 0 1 ...
 $ IQ        : int  82 79 93 87 103 77 88 96 89 109 ...
 $ NormIQ    : num [1:10000, 1] -1.23 -1.431 -0.49 -0.894 0.181 ...
  ..- attr(*, "scaled:center")= num 100
  ..- attr(*, "scaled:scale")= num 14.9


In [3]:
model1 = lm(LogEarning~College,data=school)
summary(model1)


Call:
lm(formula = LogEarning ~ College, data = school)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.6224  -1.8472  -0.0001   1.8487   9.8909 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  9.49699    0.03881   244.7   <2e-16 ***
College      2.41034    0.05453    44.2   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.726 on 9998 degrees of freedom
Multiple R-squared:  0.1635,	Adjusted R-squared:  0.1634 
F-statistic:  1954 on 1 and 9998 DF,  p-value: < 2.2e-16


Estimated Treatment Effect is: people who go to college is expected to earn 241.034% more than people who didn't go to college, provided that everything else is the same. This is not a good estimation, because College degree and earning are both positively correlated with individual abilities(IQ). Therefore, this will be an Overestimation. 

### (B).

Assume that ability is reflected by IQ:

In [4]:
model2 = lm(LogEarning ~ College+NormIQ,data=school)
summary(model2)



Call:
lm(formula = LogEarning ~ College + NormIQ, data = school)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.1309  -1.8188   0.0341   1.8039  10.3312 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.15444    0.05185  195.83   <2e-16 ***
College      1.11258    0.08763   12.70   <2e-16 ***
NormIQ       0.82018    0.04381   18.72   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.68 on 9997 degrees of freedom
Multiple R-squared:  0.1918,	Adjusted R-squared:  0.1917 
F-statistic:  1186 on 2 and 9997 DF,  p-value: < 2.2e-16


Estimated Treatment Effect is: people who go to college is expected to earn 111.258% more than people who didn't go to college, provided that IQ and everything else is the same.. The rational is to control the IQ of the samples, and adjust only "College". 

### (C).

In [5]:
model3 = lm(LogEarning~College + NormIQ + College* NormIQ,data=school)
summary(model3)


Call:
lm(formula = LogEarning ~ College + NormIQ + College * NormIQ, 
    data = school)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.0544  -1.8158   0.0361   1.7935  10.4077 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)    10.04210    0.06355 158.008  < 2e-16 ***
College         1.12524    0.08769  12.832  < 2e-16 ***
NormIQ          0.68004    0.06343  10.722  < 2e-16 ***
College:NormIQ  0.26780    0.08768   3.054  0.00226 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.679 on 9996 degrees of freedom
Multiple R-squared:  0.1926,	Adjusted R-squared:  0.1923 
F-statistic: 794.7 on 3 and 9996 DF,  p-value: < 2.2e-16


Estimated Treatment Effect is: people who go to college is expected to earn (112.524 + 26.78 * NormIQ)% more than people who didn't go to college, provided that IQ and everything else is the same.

### (D).

* The treatment group and the control group are the same with regard to all the other covariates

* Random assignment to the treatment and control groups.

* Random sampling from the population

* The unobservables are also balanced.

### (E).

In [6]:
# balance check
BalanceCheck1 = lm(LogEarning~ NormIQ,data=school )
summary(BalanceCheck1)


Call:
lm(formula = LogEarning ~ NormIQ, data = school)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.1121  -1.8343   0.0152   1.8438  10.8003 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.71807    0.02701  396.81   <2e-16 ***
NormIQ       1.26025    0.02701   46.66   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.701 on 9998 degrees of freedom
Multiple R-squared:  0.1788,	Adjusted R-squared:  0.1787 
F-statistic:  2177 on 1 and 9998 DF,  p-value: < 2.2e-16


In [7]:
library('Matching')
library(caTools)

Warning message:
"package 'Matching' was built under R version 3.5.3"Loading required package: MASS
Warning message:
"package 'MASS' was built under R version 3.5.3"## 
##  Matching (Version 4.9-5, Build Date: 2019-03-05)
##  See http://sekhon.berkeley.edu/matching for additional documentation.
##  Please cite software as:
##   Jasjeet S. Sekhon. 2011. ``Multivariate and Propensity Score Matching
##   Software with Automated Balance Optimization: The Matching package for R.''
##   Journal of Statistical Software, 42(7): 1-52. 
##

Warning message:
"package 'caTools' was built under R version 3.5.3"

In [10]:

# Fit a logistic regression to estimate the propensity score
pscore_model = glm(College ~ IQ, data=school, family=binomial)

pscore = predict(pscore_model,type="response")


school$pscore = pscore
LogEarning = school$LogEarning
College = school$College

#Run propensity score matching and estimate the average treatment effect on treated
match.pscore = Match(Y =LogEarning, Tr=College, X=pscore, estimand = "ATT", M = 1)
summary(match.pscore)

Warning message:
"glm.fit: fitted probabilities numerically 0 or 1 occurred"


Estimate...  -0.65846 
AI SE......  1.7046 
T-stat.....  -0.38629 
p.val......  0.69928 

Original number of observations..............  10000 
Original number of treated obs...............  5066 
Matched number of observations...............  5066 
Matched number of observations  (unweighted).  79019 



In [11]:
##Balance Check
MatchBalance(College ~ IQ,data=school,match.out=match.pscore)


***** (V1) IQ *****
                       Before Matching 	 	 After Matching
mean treatment........     111.92 	 	     111.92 
mean control..........      88.37 	 	     104.24 
std mean diff.........     254.52 	 	     82.965 

mean raw eQQ diff.....     23.544 	 	    0.83228 
med  raw eQQ diff.....         22 	 	          0 
max  raw eQQ diff.....         49 	 	         49 

mean eCDF diff........    0.22424 	 	  0.0079998 
med  eCDF diff........   0.079945 	 	  0.0019489 
max  eCDF diff........    0.92492 	 	   0.045635 

var ratio (Tr/Co).....      1.069 	 	     32.926 
T-test p-value........ < 2.22e-16 	 	 < 2.22e-16 
KS Bootstrap p-value.. < 2.22e-16 	 	 < 2.22e-16 
KS Naive p-value...... < 2.22e-16 	 	 < 2.22e-16 
KS Statistic..........    0.92492 	 	   0.045635 



## Question 3

### (A).

In [2]:
newFeature = read.csv("D:/Study/Business Analytics/LA/HW7/NewFeature.csv")
str(newFeature)

'data.frame':	10000 obs. of  3 variables:
 $ Satisfaction : int  10 2 6 10 6 4 7 3 4 9 ...
 $ Adoption     : int  1 0 0 1 0 0 1 0 0 1 ...
 $ Encouragement: int  1 0 0 0 0 1 0 1 0 0 ...


In [4]:
model1 =lm(Satisfaction~Adoption, data = newFeature)
summary(model1)


Call:
lm(formula = Satisfaction ~ Adoption, data = newFeature)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.1261 -1.1261 -0.1261  0.8739  4.3488 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.65121    0.01935   240.4   <2e-16 ***
Adoption     3.47492    0.02863   121.4   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.426 on 9998 degrees of freedom
Multiple R-squared:  0.5958,	Adjusted R-squared:  0.5957 
F-statistic: 1.473e+04 on 1 and 9998 DF,  p-value: < 2.2e-16


$\hat{\beta_{A}}$ is 3.47492. This is not a good estimation, because people who are already satisfied with facebook tend to be more willing to adopt new features. As 'Satisfaction' is positively correlated with 'Adoption', and 'Adoption' is positively correlated with 'Encouragement', this would be an overeestimation.

### (B).

Because encouragement is correlated with adoption, but not correlated to the error term (Bias).

### (C).

In [14]:
#first-stage regression X~Z and compute X_hat
model_fs = lm(Adoption~Encouragement,data=newFeature)
newFeature$Adoption1 = predict(model_fs,newdata=newFeature)
summary(model_fs)
##F-test is significant, so the strong first-stage assumption holds.


Call:
lm(formula = Adoption ~ Encouragement, data = newFeature)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5685 -0.3445 -0.3445  0.4315  0.6555 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)   0.344489   0.006872   50.13   <2e-16 ***
Encouragement 0.223974   0.009708   23.07   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4854 on 9998 degrees of freedom
Multiple R-squared:  0.05054,	Adjusted R-squared:  0.05045 
F-statistic: 532.2 on 1 and 9998 DF,  p-value: < 2.2e-16


In [15]:
#second-stage regression Y~X_hat
model_ss  = lm(Satisfaction~Adoption1,data=newFeature)
summary(model_ss)


Call:
lm(formula = Satisfaction ~ Adoption1, data = newFeature)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.3397 -1.3397 -0.1363  1.6603  3.8637 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.82335    0.09408   61.90  < 2e-16 ***
Adoption1    0.90836    0.20007    4.54 5.68e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.24 on 9998 degrees of freedom
Multiple R-squared:  0.002058,	Adjusted R-squared:  0.001958 
F-statistic: 20.61 on 1 and 9998 DF,  p-value: 5.684e-06


In [13]:
library("AER")
#2sls combined estimation
model2 = ivreg(Satisfaction~Adoption|Encouragement,data=newFeature)
summary(model2)


Call:
ivreg(formula = Satisfaction ~ Adoption | Encouragement, data = newFeature)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.8234 -0.8234  0.1766  1.2683  3.2683 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.82335    0.08042  72.412  < 2e-16 ***
Adoption     0.90836    0.17102   5.311 1.11e-07 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.915 on 9998 degrees of freedom
Multiple R-Squared: 0.2708,	Adjusted R-squared: 0.2707 
Wald test: 28.21 on 1 and 9998 DF,  p-value: 1.112e-07 


$\hat{\gamma_{A}}$ is 0.90836. It means adopting the new feature causes an average 0.90836 units increase in satisfaction